src/mono/mono/mini/mini-ops.h

```diff
@@ -763,6 +763,7 @@ MINI_OP(OP_NEWARR, "newarr", IREG, IREG, NONE)
 MINI_OP(OP_LDLEN, "ldlen", IREG, IREG, NONE)
 /* inst_p0 is the exception name to throw or NULL */
 MINI_OP(OP_BOUNDS_CHECK, "bounds_check", NONE, IREG, IREG)
+MINI_OP(OP_BOUNDS_CHECK_SEXT, "bounds_check_sext", NONE, IREG, IREG)
```

Comment:
Using OP_BOUNDS_CHECK for both and storing the 'sext' information in ins->inst_c0 or something could simplify some code.

Reply:
added the flag inside `ins->backend` since both slots in `ins->data` were used by the instruction